numpy/polynomial/polyutils.py

```diff
@@ -459,3 +459,51 @@ def _vander3d(vander_f, x, y, z, deg):
     vz = vander_f(z, degz)
     v = vx[..., None, None]*vy[..., None,:, None]*vz[..., None, None,:]
     return v.reshape(v.shape[:-3] + (-1,))
+
+
+def _valnd(val_f, c, *args):
+    """
+    Helper function used to implement the ``<type>val<n>d`` functions.
+
+    Parameters
+    ----------
+    val_f : function(array_like, array_like, tensor: bool) -> array_like
+        The ``<type>val`` function, such as ``polyval``
+    c, args :
+        See the ``<type>val<n>d`` functions for more detail
+    """
+    try:
+        args = tuple(np.array(args, copy=False))
```

Comment:
Future work - probably use `args = np.broadcast_arrays(*args)` here instead, now that it's in one place.

Reply:
Good thought.